lib/rubocop/cop/factory_bot/factory_association_with_strategy.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module FactoryBot
+      # Use definition in factory association instead of hard coding a strategy.
+      #
+      # @example
+      #   # bad - only works for one strategy
+      #   factory :foo do
+      #     profile { create(:profile) }
+      #   end
+      #
+      #   # good - implicit
+      #   factory :foo do
+      #     profile
+      #   end
+      #
+      #   # good - explicit
+      #   factory :foo do
+      #     association :profile
+      #   end
+      #
+      #   # good - inline
+      #   factory :foo do
+      #     profile { association :profile }
+      #   end
+      #
+      class FactoryAssociationWithStrategy < ::RuboCop::Cop::Base
+        MSG = 'Use an implicit, explicit or inline definition instead of ' \
+              'hard coding a strategy for setting association within factory.'
+
+        HARDCODED = Set.new([:create, :build, :build_stubbed])
+
+        # @!method factory_declaration(node)
+        def_node_matcher :factory_declaration, <<~PATTERN
+          (block (send nil? {:factory :trait} ...)
+            ...
+          )
+        PATTERN
+
+        # @!method factory_strategy_association(node)
+        def_node_matcher :factory_strategy_association, <<~PATTERN
```

Comment:
@pirj is there a more elegant way to do it?  
This is the way I found to make this [test](https://github.com/rubocop/rubocop-factory_bot/pull/34/files#diff-6726f09e7ef0a2e1d32383d4cd1caeddeb973a4be6d9e2c3691ec0325e4f98d6R49-R62) pass.

Reply:
It is as elegant as a node pattern can be 😅
Seriously, this construct that looks clumsy at first, replaces several lines of Ruby code, and the code that transpiles it into Ruby is optimised and known to have no side effects.

Reply:
Haha sorry my comment was not clear at all 🙈 

I was more talking about using two node matcher and used in two 3 nested blocks
https://github.com/rubocop/rubocop-factory_bot/pull/34/files#diff-10eb8e610da9d71b102d2bd7dad3e70e381264a4ff3fe0cb43b1c4085c2d8ae3R51-R59